### PR TITLE
Translate JavaScript chapter to Portuguese

### PR DESCRIPTION
**Pull request by HakaCode**
forgive me for the delay in the translation.
Makes progress on #505

### CODE REVIEW

File: src/content/pt/2019/javascript.md [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+Para prefetch, temos o seguinte:
+
+<figure>
+   <a href="/static/images/2019/javascript/fig16.png">
+      <img src="/static/images/2019/javascript/fig16.png" alt="Figura 16. Porcentagem de sites usando rel=prefetch para scripts." aria-labelledby="fig16-caption" aria-describedby="fig16-description" width="600" height="371" data-width="600" data-height="371" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=547807937&format=interactive">
+   </a>
+   <div id="fig16-description" class="visually-hidden">Gráfico de barras mostrando 0,08% dos sites para desktop usam 'rel=prefetch', e 0,08% dos sites para celular.</div>
+   <figcaption id="fig16-caption">Figura 16. Porcentagem de sites usando rel=prefetch para scripts.</figcaption>
+</figure>
```
Comment (bazzadp) [inline src/content/pt/2019/javascript.md:380]:
```suggestion
{{ figure_markup(
  image="/static/images/2019/javascript/fig16.png",
  caption="Porcentagem de sites usando rel=prefetch para scripts.",
  description="Gráfico de barras mostrando 0,08% dos sites para desktop usam 'rel=prefetch', e 0,08% dos sites para celular.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=547807937&format=interactive"
  )
}}
```

File: src/content/pt/2019/javascript.md [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+Para dispositivos móveis e desktops, 0,08% das páginas aproveitam a pré-busca para qualquer um de seus scripts.
+
+## Novas APIs
+
+JavaScript continua a evoluir como linguagem. A cada ano uma nova versão do padrão da linguagem, conhecida como ECMAScript, é lançada com novas APIs e recursos que passam pelas etapas da proposta de se tornar parte da própria linguagem.
+
+Com o HTTP Archive, podemos dar uma olhada em qualquer API mais recente que seja suportada (ou prestes a ser) e ver o quão difundida ela está em uso. Essas APIs já podem ser usadas em navegadores que as suportam _ou_ com um polyfill anexado para garantir que ainda funcionem para todos os usuários.
+
+**Quantos sites usam as seguintes APIs?**
```
Comment (bazzadp) [inline src/content/pt/2019/javascript.md:390]:
```suggestion
Quantos sites usam as seguintes APIs?
```

File: src/content/pt/2019/javascript.md [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+
+- [Atomics](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics)
+- [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+- [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)
+- [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)
+- [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)
+- [WeakSet](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet)
```
Comment (bazzadp) [inline src/content/pt/2019/javascript.md:397]:
```suggestion
- [Atomics](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Atomics)
- [Intl](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Intl)
- [Proxy](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Proxy)
- [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)
- [WeakMap](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)
- [WeakSet](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/WeakSet)
```

File: src/content/pt/2019/javascript.md [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+
+<figure>
+   <a href="/static/images/2019/javascript/fig17.png">
+      <img src="/static/images/2019/javascript/fig17.png" alt="Figura 17. Uso de novas APIs JavaScript." aria-labelledby="fig17-caption" aria-describedby="fig17-description" width="600" height="371" data-width="600" data-height="371" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=594315296&format=interactive">
+   </a>
+   <div id="fig17-description" class="visually-hidden">Gráfico de barras mostrando 25,5% / 36,2% dos sites em computadores e dispositivos móveis usam WeakMap, 6,1% / 17,2% usam WeakSet, 3.9%/14.0% usa Intl, 3.9%/4.4% usa Proxy, 0.4%/0.4% usa Atomics, e 0.2%/0.2% usa SharedArrayBuffer.</div>
+   <figcaption id="fig17-caption">Figura 17. Uso de novas APIs JavaScript.</figcaption>
+</figure>
```
Comment (bazzadp) [inline src/content/pt/2019/javascript.md:405]:
```suggestion
{{ figure_markup(
  image="/static/images/2019/javascript/fig17.png",
  caption="Uso de novas APIs JavaScript.",
  description="Gráfico de barras mostrando 25,5% / 36,2% dos sites em computadores e dispositivos móveis usam WeakMap, 6,1% / 17,2% usam WeakSet, 3.9%/14.0% usa Intl, 3.9%/4.4% usa Proxy, 0.4%/0.4% usa Atomics, e 0.2%/0.2% usa SharedArrayBuffer.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=594315296&format=interactive"
  )
}}
```

File: src/content/pt/2019/javascript.md [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+
+Embora úteis, existem vários motivos pelos quais muitos sites podem não querer incluir mapas de origem em seu site de produção final, como a escolha de não expor o código-fonte completo ao público. Então, quantos sites realmente incluem mapas de origem?
+
+<figure>
+   <a href="/static/images/2019/javascript/fig18.png">
+      <img src="/static/images/2019/javascript/fig18.png" alt="Figura 18. Porcentagem de sites usando mapas de origem." aria-labelledby="fig18-caption" aria-describedby="fig18-description" width="600" height="371" data-width="600" data-height="371" data-seamless data-frameborder="0" data-scrolling="no" data-iframe="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=906754154&format=interactive">
+   </a>
+   <div id="fig18-description" class="visually-hidden">Gráfico de barras mostrando 18% dos sites para desktop e 17% dos sites para celular usam mapas de origem.</div>
+   <figcaption id="fig18-caption">Figura 18. Porcentagem de sites usando mapas de origem.</figcaption>
+</figure>
```
Comment (bazzadp) [inline src/content/pt/2019/javascript.md:425]:
```suggestion
{{ figure_markup(
  image="/static/images/2019/javascript/fig18.png",
  caption="Porcentagem de sites usando mapas de origem.",
  description="Gráfico de barras mostrando 18% dos sites para desktop e 17% dos sites para celular usam mapas de origem.",
  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpzDb9HGbdVvin6YPTOmw11qBVGGysltxmH545fUfnqIThAq878F_b-KxUo65IuXaeFVSnlmJ5K1Dm/pubchart?oid=906754154&format=interactive"
  )
}}
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Looks pretty good. A few things we should change though:
- We moved to new [figure markup very recently](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1236)
- You've bolded a number of sentences but not sure why?
- Translated versions of MDN

File: src/content/pt/2019/javascript.md [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+   </a>
+   <div id="fig9-description" class="visually-hidden">Gráfico de barras mostrando 0/17 bytes de JavaScript sendo baixados para dispositivos móveis para conteúdo de origem e conteúdo de terceiros, respectivamente, no 10º percentil, 6/54 no 25º percentil, 83/217 no 50º percentil, 189/477 no 75º percentil e 380/827 no 90º percentil.</div>
+   <figcaption id="fig9-caption">Figura 9. Distribuição do JavaScript total baixado em dispositivos móveis.</figcaption>
+</figure>
+
+Na média, 89% mais código de conteúdo de terceiros é usado do que código de conteúdo de origem criado pelo desenvolvedor para desktops e dispositivos móveis. Isso mostra claramente que o código de terceiros pode ser um dos maiores contribuintes para a inflação. Para obter mais informações sobre o impacto de terceiros, consulte o capítulo ["Terceiros"](./third-parties).
```
[the remark this answers is not in the document]
Comment (bazzadp) [reply src/content/pt/2019/javascript.md:133]: Actually see it is slightly different in English too so closing this.

File: src/content/pt/2019/javascript.md [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+  )
+}}
+
+A cada percentil, os tempos de processamento são mais longos para páginas da web para celular do que para desktop. A mediana do tempo total do thread principal no desktop é de 849 ms, enquanto no celular está em um número maior: 2.437 ms.
+
+Embora esses dados mostrem quanto tempo pode levar para um dispositivo móvel processar JavaScript em comparação com uma máquina de desktop mais poderosa, os dispositivos móveis também variam em termos de capacidade de computação. O gráfico a seguir mostra como os tempos de processamento em uma única página da web podem variar significativamente dependendo da classe do dispositivo móvel.
+
+{{ figure_markup(
+  image="/static/images/2019/javascript/js-processing-reddit.png",
+  caption="Tempos de processamento de JavaScript para reddit.com. De <a href="https://v8.dev/blog/cost-of-javascript-2019">O custo do JavaScript em 2019</a>.",
```
Comment (bazzadp) [inline src/content/pt/2019/javascript.md:72]:
```suggestion
  caption='Tempos de processamento de JavaScript para reddit.com. De <a href="https://v8.dev/blog/cost-of-javascript-2019">O custo do JavaScript em 2019</a>.',
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Almost there now!

Can your add yourself to the `src/config/2019.json` file? It's in alphabetical order first name so add yourself to line 1059 or thereabouts?

Also note that we can't actually send this language (and this chapter) live until we translate `templates/en/2019/featured_chapters.html` - could you tackle that one next? Then we can release this.

**Comment by HakaCode** [comment]
> Can your add yourself to the `src/config/2019.json` file? It's in alphabetical order first name so add yourself to line 1059 or thereabouts?

read my mind.
I would ask that today.

> Also note that we can't actually send this language (and this chapter) live until we translate `templates/en/2019/featured_chapters.html` - could you tackle that one next? Then we can release this.

Okay, I'm in no hurry, but I will still resolve this as soon as possible so that it is released to the community.

**Review by HakaCode — COMMENTED** [review]
ready for the merge!

**Review by bazzadp — APPROVED** [review]

**Pull request merged**